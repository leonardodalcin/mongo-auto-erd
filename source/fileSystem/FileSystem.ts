import * as fs from 'fs'
import * as path from 'path'
export class FileSystem {
  public static writeObjToFile(filename: string, obj: any) {
    fs.writeFileSync(path.resolve(filename), JSON.stringify(obj))
  }
}
