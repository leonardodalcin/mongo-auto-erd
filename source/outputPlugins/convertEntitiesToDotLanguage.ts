import { IEntity } from '@interfaces/IEntity'
import { IPropertyType } from '@interfaces/IPropertyType'
import { digraph } from 'graphviz'
import { resolve } from 'path'

export function convertEntitiesToDotLanguage(entities: IEntity[]) {
  const graph = digraph('1')
  for (const entity of entities) {
    const mapped: { [propName: string]: IPropertyType } = {}
    for (const property of entity.properties) {
      mapped[property.name] = property.types[0]
    }
    graph.addNode(entity.name, { label: JSON.stringify(mapped, undefined, 2).replace(/"/g, '') })

  }
  graph.output('png', resolve(__dirname + '.png'))
  return graph.to_dot()
}
