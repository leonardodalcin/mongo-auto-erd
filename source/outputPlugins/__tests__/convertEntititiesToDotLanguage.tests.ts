import { FileSystem } from '@fileSystem/FileSystem'
import { convertEntitiesToDotLanguage } from '@outputPlugins/convertEntitiesToDotLanguage'
import { ERD } from './ERD.mock'

describe('convertEntitiesToDotLanguage.', () => {
  it('should convert mock to a valid dot file',  () => {
    const dotLanguageGraph = convertEntitiesToDotLanguage(ERD)
    expect(dotLanguageGraph).toBeTruthy()
    FileSystem.writeObjToFile('test.dot', dotLanguageGraph)
  })
})
