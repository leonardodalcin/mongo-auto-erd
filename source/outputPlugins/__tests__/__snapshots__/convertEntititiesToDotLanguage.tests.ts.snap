// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertEntitiesToDotLanguage. should convert mock to a valid dot file 1`] = `
"digraph 1 {
  \\"users\\" [ shape =record, label = \\"{users | __v : number\\\\l_id : objectId\\\\lname : string}\\" ];
  \\"photos\\" [ shape =record, label = \\"{photos | __v : number\\\\l_id : objectId\\\\ltitle : string\\\\luser : objectId}\\" ];
  \\"usergroups\\" [ shape =record, label = \\"{usergroups | __v : number\\\\l_id : objectId\\\\lname : string\\\\lusers : objectId}\\" ];
  \\"users\\" -> \\"users\\";
  \\"photos\\" -> \\"photos\\";
  \\"photos\\" -> \\"users\\";
  \\"usergroups\\" -> \\"usergroups\\";
  \\"usergroups\\" -> \\"users\\";
}
"
`;
